### ISSUE TITLE
Error on non divisible users/agents

### ISSUE THREAD

**Issue by mocenas** [Hyperfoil/Hyperfoil#86]
On Openshift controller, when using phase type "always" with parameter users e.g. 51 and using two agents, upload of scenario fails with error.
```
ERROR: Cannot slice phase listing, property users cleanly: 51 * 0.5 is not an integer.
Failed to parse the benchmark.
```
Does really the users count have to be divisible by number of agents? This error doesn't seem necessary.

**Comment by rvansa** [comment]
Hmm, it's true that we don't have to insist on exactly equal distribution of load to all agents. There's a technical challenge that we would have to either distribute different instances of benchmark to each agent, or perform any distribution only at the agents after these receive the benchmark.

**Comment by rvansa** [comment]
Related concern (for record): when agent nodes have different number of cores, it would be beneficial to use uneven executor count.

**Comment by rvansa** [comment]
Implemented in 12c2989817725df362fd349c1a615a6375a2aa95 (`always.users` and similar do not have to be divisible by #agents).
Agents can now define property `threads` which overrides the default defined as top-level attribute in the benchmark. Users/rate/sessions/connections will be divided based on the number of actual threads in each agent.
